### PR TITLE
Gender backwards compatibility bug

### PR DESCRIPTION
**Pull request by anwzhang**
This PR include a fix for a bug that when no SDK value passes in or version 0.0.0 passes in, User object parsing should not be handled.

### CODE REVIEW

File: pbs/pbsrequest.go [hunk 1]
```diff
@@ -328,3 +332,72 @@ func (p PBSRequest) String() string {
...
+		} else if r > l {
+			return 1
+		}
+	}
+
+	return 0
+
+}
+
+var specialForms = map[string]int{
```
Comment (dbemiller) [inline pbs/pbsrequest.go]: I'm a little confused by these... is there some reason we can't stick to standard semantic versioning?

If we do, there are a bunch of libraries in Go which already do this for you. For example:

https://github.com/blang/semver
https://github.com/hashicorp/go-version

**Comment by mkendall07** [comment]
what's the impact for this bug?

File: pbs/pbsrequest.go (hunk 1 above)
Comment (anwzhang) [reply pbs/pbsrequest.go]: Will check the libraries.

**Comment by anwzhang** [comment]
@mkendall07 
Early versions of prebid mobile are sending requests with gender indicated by numbers, it can't be parsed by latest Prebid Server after the change of gender to use string. To handle those traffic, adding a check here to ignore the sent gender.

File: pbs/pbsrequest_test.go [hunk 2]
```diff
@@ -450,3 +450,91 @@ func TestParseMobileRequest(t *testing.T) {
 		t.Errorf("Parse sdk platform failed")
 	}
 }
+
+func TestParseMalformedMobileRequest(t *testing.T) {
+	body := []byte(`{
```
Comment (nhedley) [inline pbs/pbsrequest_test.go]: What's the difference between this test and the earlier test `TestParseMobileRequestFirstVersion`? They look the same to me

Comment (anwzhang) [reply pbs/pbsrequest_test.go]: No SDK version is passed in this request. The sample request client sent did not have SDK version, that's a separate issue to investigate but this test case is to cover that.

**Comment by dbemiller** [comment]
> Early versions of prebid mobile are sending requests with gender indicated by numbers, it can't be parsed by latest Prebid Server after the change of gender to use string so clients using early versions can't be monetized. To handle those traffic, adding a check here to ignore the sent gender.

This would make a good code comment ^^. Changes look fine in either case though.

File: pbs/pbsrequest.go [hunk 3]
```diff
@@ -13,6 +13,7 @@ import (
 	"github.com/spf13/viper"
 	"golang.org/x/net/publicsuffix"
 
+	"github.com/blang/semver"
```
Comment (mkendall07) [inline pbs/pbsrequest.go]: Probably should add this to glide.yaml

Comment (dbemiller) [reply pbs/pbsrequest.go]: oh man... yes. This is essential before merge

**Comment by anwzhang** [comment]
@dbemiller glide.lock already contains blang/semver so only updated glide.yaml.

**Pull request merged**